nixos/modules/services/misc/plikd.nix

```diff
@@ -0,0 +1,84 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.plikd;
+
+  format = pkgs.formats.toml {};
+
+  plikdCfg = format.generate "plikd.cfg" cfg.settings;
+in
+{
+  options = {
+    services.plikd = {
+      enable = mkEnableOption "Enable the plikd server";
+
+      user = mkOption {
+        type = types.str;
+        default = "plikd";
+        description = "User account under which Plikd runs.";
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = {
+          ListenPort    = 8080;
+          ListenAddress = "0.0.0.0";
+          DataBackend   = "file";
+          DataBackendConfig = {
+            Directory = "/var/lib/plikd";
+          };
+          MetadataBackendConfig = {
+            Driver = "sqlite3";
+            ConnectionString = "/var/lib/plikd/plik.db";
+          };
+        };
+        description = "Server configuration";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "plikd";
+        description = "Group account under which Plikd runs.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.plikd;
+        defaultText = "pkgs.plikd";
+        description = "The plikd package to use.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.plikd = {
```

Comment:
After we merge this PR you might consider contributing this `systemd` unit to upstream. I'm sure they would benefit from your contribution.